java-frontend/src/main/java/org/sonar/java/se/xproc/ExceptionalYield.java

```diff
@@ -57,30 +60,46 @@ public ExceptionalYield(ExplodedGraph.Node node, MethodBehavior behavior) {
       .distinct();
   }
 
-  public void setExceptionType(String exceptionType) {
+  public void setExceptionType(@Nullable Type exceptionType) {
     this.exceptionType = exceptionType;
+    this.exceptionTypeName = null;
+  }
+
+  public void setExceptionTypeName(String exceptionTypeName) {
+    this.exceptionTypeName = exceptionTypeName;
+    this.exceptionType = null;
   }
 
   public Type exceptionType(SemanticModel semanticModel) {
-    if (exceptionType == null) {
-      return Symbols.unknownType;
+    if (exceptionType != null) {
+      return exceptionType;
+    }
+    if (exceptionTypeName != null) {
+      Type type = semanticModel.getClassType(this.exceptionTypeName);
+      return type == null ? Symbols.unknownType : type;
```

Comment:
I think that if it's not null, we should then store the type new type as being `exceptionType`, so we don't need to recompute it later. Maybe we could also directly computing it when setting the exceptionTypeName as well.